File: ViewBindingJava/app/src/main/java/com/websarva/wings/android/viewbindingjava/MainActivity.java
package com.websarva.wings.android.viewbindingjava;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.websarva.wings.android.viewbindingjava.databinding.ActivityMainBinding;
import com.websarva.wings.android.viewbindingjava.ui.MainViewModel;

public class MainActivity extends AppCompatActivity {
	private MainViewModel _mainViewModel;
	private ActivityMainBinding _activityMainBinding;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		_activityMainBinding = ActivityMainBinding.inflate(getLayoutInflater());
		View contentView = _activityMainBinding.getRoot();
		setContentView(contentView);

		ViewModelProvider provider = new ViewModelProvider(MainActivity.this);
		_mainViewModel = provider.get(MainViewModel.class);

		_activityMainBinding.tvRand.setText(_mainViewModel.getRandNumStr());
		_activityMainBinding.btnRand.setOnClickListener(new ButtonClickListener());

//		TextView tvRand = findViewById(R.id.tvRand);
//		tvRand.setText(_mainViewModel.getRandNumStr());
//		Button btnRand = findViewById(R.id.btnRand);
//		btnRand.setOnClickListener(new ButtonClickListener());
	}

	private class ButtonClickListener implements View.OnClickListener {
		@Override
		public void onClick(View v) {
			_mainViewModel.generateNewRandNum();
			_activityMainBinding.tvRand.setText(_mainViewModel.getRandNumStr());
		}
	}
}

File: ViewBindingKotlin/app/src/main/java/com/websarva/wings/android/viewbindingkotlin/MainActivity.kt
package com.websarva.wings.android.viewbindingkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.activity.viewModels
import com.websarva.wings.android.viewbindingkotlin.databinding.ActivityMainBinding
import com.websarva.wings.android.viewbindingkotlin.ui.MainViewModel

class MainActivity : AppCompatActivity() {
	private val _mainViewModel by viewModels<MainViewModel>()
	private lateinit var _activityMainBinding: ActivityMainBinding

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		_activityMainBinding = ActivityMainBinding.inflate(layoutInflater)
		setContentView(_activityMainBinding.root)

		_activityMainBinding.tvRand.text = _mainViewModel.getRandNumStr()
		_activityMainBinding.btnRand.setOnClickListener(ButtonClickListener())

//		val tvRand = findViewById<TextView>(R.id.tvRand)
//		tvRand.text = _mainViewModel.getRandNumStr()
//		val btnRand = findViewById<Button>(R.id.btnRand)
//		btnRand.setOnClickListener(ButtonClickListener())
	}

	private inner class ButtonClickListener : View.OnClickListener {
		override fun onClick(view: View?) {
			_mainViewModel.generateNewRandNum()
			_activityMainBinding.tvRand.text = _mainViewModel.getRandNumStr()
		}
	}
}
